src/hotspot/share/gc/shenandoah/shenandoahControlThread.cpp

```diff
@@ -401,7 +380,11 @@ void ShenandoahControlThread::service_stw_degenerated_cycle(GCCause::Cause cause
   gc.collect(cause);
 }
 
-bool ShenandoahControlThread::is_explicit_gc(GCCause::Cause cause) const {
+bool ShenandoahControlThread::should_run_full_gc(GCCause::Cause cause) {
+  return is_explicit_gc(cause) ? !ExplicitGCInvokesConcurrent : !ShenandoahImplicitGCInvokesConcurrent;
+}
```

Comment:
This sounds like a policy decision, which means a good fit for it is in `ShenandoahCollectorPolicy`?

Reply:
Agree. Moved it in there.